spec.html

```diff
@@ -20653,6 +21265,239 @@ <h1>Static Semantics: PrototypePropertyNameList</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-class-definitions-static-semantics-allprivateidentifiersvalid" type="sdo" aoid="AllPrivateIdentifiersValid">
+      <h1>Static Semantics: AllPrivateIdentifiersValid</h1>
+      <p>With parameter _names_.</p>
+
+      <p>Every grammar production alternative in this specification which is not listed below implicitly has the following default definition of AllPrivateIdentifiersValid:</p>
+      <emu-alg>
+        1. For each child node _child_ of this Parse Node, do
+          1. If _child_ is an instance of a nonterminal, then
+            1. If AllPrivateIdentifiersValid of _child_ with argument _names_ is *false*, return *false*.
+        1. Return *true*.
+      </emu-alg>
+
+      <emu-grammar>MemberExpression : MemberExpression `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, return *true*.
```

Comment:
```suggestion
        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
          1. Return AllPrivateIdentifiersValid of |MemberExpression| with argument _names_.
```

(Similarly for CallExpression and OptionalChain.)

Reply:
Done in https://github.com/tc39/ecma262/pull/1668/commits/d859102ee4616af69b588d753ffca20087c12168.